src/model-helpers/Profile.js

```diff
@@ -0,0 +1,164 @@
+  return { profileInfo, profileDataset, profileThing };
+};
+
+/**
+ * A function that updates the user's profile information based on what is inputed
+ * in the Profile page
+ *
+ * @function updateProfileInfo
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {object} profileData - The object containing the information related
+ * to the person on their profile card, the profile dataset, and the profile Thing
+ * @param {object} inputValues - The inputs for updating profile information
+ * on their profile card
+ * @returns {Promise} Promise - Performs action to update profile card on the
+ * user's profile card
+ */
+export const updateProfileInfo = async (session, profileData, inputValues) => {
+  let { profileDataset, profileThing } = profileData;
+  const { profileInfo } = profileData;
+
+  Object.keys(inputValues).forEach((input) => {
+    switch (inputValues[input]) {
+      case null:
+        return;
+      case '':
+        profileThing = removeStringNoLocale(
+          profileThing,
+          RDF_PREDICATES[input],
+          profileInfo[input]
+        );
+        break;
+      default:
+        profileThing = buildThing(profileThing)
+          .setStringNoLocale(RDF_PREDICATES[input], inputValues[input])
+          .build();
+        break;
+    }
+  });
+
+  profileDataset = setThing(profileDataset, profileThing);
+
+  await saveSolidDatasetAt(session.info.webId, profileDataset, { fetch: session.fetch });
+};
+
+/**
+ * Function that upload profile image of user onto their profile container and
+ * reference it on their Pod
+ *
+ * @function uploadProfileImage
+ * @param {Session} session - Solid's Session Object {@link Session}
+ * @param {object} profileData - The object containing the information related
+ * to the person on their profile card, the profile dataset, and the profile Thing
+ * @param {Blob} inputImage - File blob being uploaded for profile image
+ * @returns {Promise} Promise - Performs a file upload of your profile image onto
+ * your pod at /profile
+ */
+export const uploadProfileImage = async (session, profileData, inputImage) => {
+  let { profileDataset, profileThing } = profileData;
+
+  const profileContainer = `${session.info.webId.split('card')[0]}`;
+  let profileImgFilename = inputImage.name.replace(' ', '%20');
+
+  if (profileImgFilename.endsWith('.jpg')) {
+    profileImgFilename = profileImgFilename.replace('.jpg', '.jpeg');
+  }
+
+  await saveFileInContainer(profileContainer, inputImage, {
+    slug: profileImgFilename,
+    contentType: inputImage.type,
+    fetch: session.fetch
+  });
+
+  profileThing = buildThing(profileThing)
+    .addUrl(RDF_PREDICATES.profileImg, `${profileContainer}${profileImgFilename}`)
+    .build();
+
+  profileDataset = setThing(profileDataset, profileThing);
+
+  await saveSolidDatasetAt(session.info.webId, profileDataset, { fetch: session.fetch });
+
+  // Create ACL file for profile image
+  const imageResource = await getFile(
```

Comment:
This file is returned by the `saveFileInContainer` function. You don't need to refetch it.

Reply:
Using the newly stored `savedFile` from `saveFileInContainer`

Reply:
Had planned on using `savedFile` since it seems to be the correct object to use for Inrupt's `createAcl`. Unfortunately, it seems to err out.

https://github.com/codeforpdx/PASS/assets/14917816/9459d955-2fca-4756-adec-a8873fe78433

Thinking might be due to `saveFileInContainer` and `getFile` returns different objects and `createAcl` expects the one from `getFile`, but I'm not too sure.

![Screen Shot 2023-07-06 at 23 32 03 (2)](https://github.com/codeforpdx/PASS/assets/14917816/a6e5e88b-12ae-448d-ab4d-5db067665770)

For now, I'm returning to using `imageResource`.

Reply:
Well that's very odd, and that error message is not all that helpful. Your current implementation is fine for new then. I guess we can dig into why inrupt's libraries are misbehaving later.